armv7 (32-bit) support

There are currently some errors when building for 32-bit armv7, as mentioned at https://github.com/PureSwift/Android/pull/27. I don't know if the exclusion of this architecture was intentional or not, but there might be some simple fixes that could get it building, such as explicitly declaring `UInt32` for some types:

```
/Users/runner/work/Android/Android/.build/checkouts/Socket/Sources/Socket/System/Syscalls.swift:169:45: error: cannot convert value of type 'UInt32' to expected argument type 'socklen_t' (aka 'Int32')
167 |   //if mockingEnabled { return _mock(family, pointer, string, length) }
168 |   #endif
169 |   return inet_ntop(family, pointer, string, length)
    |                                             `- error: cannot convert value of type 'UInt32' to expected argument type 'socklen_t' (aka 'Int32')
170 | }
```

See https://github.com/swift-android-sdk/Android/actions/runs/22495334506/job/65168216274 for a build log.

--- a/.github/workflows/swift.yml
+++ b/.github/workflows/swift.yml
@@ -23,7 +23,7 @@ jobs:
     name: Linux
     strategy:
       matrix:
-        container: ["swift:6.0.3", "swift:6.1.2"]
+        container: ["swift:6.0.3", "swift:6.1.2", "swift:6.2.3"]
         config: ["debug", "release"]
         options: ["", "SWIFT_BUILD_DYNAMIC_LIBRARY=1"]
     runs-on: ubuntu-latest
@@ -41,8 +41,8 @@ jobs:
       strategy:
         fail-fast: false
         matrix:
-          swift: ['6.1', 'nightly-6.2']
-          arch: ["aarch64", "x86_64"]
+          swift: ['6.2.3', 'nightly-6.3']
+          arch: ['aarch64', 'x86_64', 'armv7']
       runs-on: macos-15
       timeout-minutes: 30
       steps:

--- a/Sources/Socket/System/CInternetAddress.swift
+++ b/Sources/Socket/System/CInternetAddress.swift
@@ -38,7 +38,7 @@ internal extension String {
                 T.family.rawValue,
                 $0,
                 cString,
-                numericCast(T.stringLength)
+                CInterop.SocketLength(T.stringLength)
             ) != nil
         }
         guard success else {

--- a/Sources/Socket/System/CInterop.swift
+++ b/Sources/Socket/System/CInterop.swift
@@ -54,6 +54,8 @@ public extension CInterop {
     typealias SocketType = CInt
     #endif
     
+    typealias SocketLength = socklen_t
+    
     /// The C `addrinfo` type
     typealias AddressInfo = addrinfo
     

--- a/Sources/Socket/System/CSocketAddress.swift
+++ b/Sources/Socket/System/CSocketAddress.swift
@@ -10,18 +10,18 @@ internal protocol CSocketAddress {
 internal extension CSocketAddress {
     
     func withUnsafePointer<Result>(
-        _ body: (UnsafePointer<CInterop.SocketAddress>, UInt32) throws -> Result
+        _ body: (UnsafePointer<CInterop.SocketAddress>, CInterop.SocketLength) throws -> Result
         ) rethrows -> Result {
         return try Swift.withUnsafeBytes(of: self) {
-            return try body($0.baseAddress!.assumingMemoryBound(to:  CInterop.SocketAddress.self), UInt32(MemoryLayout<Self>.size))
+            return try body($0.baseAddress!.assumingMemoryBound(to:  CInterop.SocketAddress.self), CInterop.SocketLength(MemoryLayout<Self>.size))
         }
     }
     
     mutating func withUnsafeMutablePointer<Result>(
-        _ body: (UnsafeMutablePointer<CInterop.SocketAddress>, UInt32) throws -> Result
+        _ body: (UnsafeMutablePointer<CInterop.SocketAddress>, CInterop.SocketLength) throws -> Result
         ) rethrows -> Result {
             return try Swift.withUnsafeMutableBytes(of: &self) {
-                return try body($0.baseAddress!.assumingMemoryBound(to:  CInterop.SocketAddress.self), UInt32(MemoryLayout<Self>.size))
+                return try body($0.baseAddress!.assumingMemoryBound(to:  CInterop.SocketAddress.self), CInterop.SocketLength(MemoryLayout<Self>.size))
         }
     }
 }

--- a/Sources/Socket/System/SocketAddress.swift
+++ b/Sources/Socket/System/SocketAddress.swift
@@ -8,7 +8,7 @@ public protocol SocketAddress: Sendable {
     
     /// Unsafe pointer closure
     func withUnsafePointer<Result, Error>(
-      _ body: (UnsafePointer<CInterop.SocketAddress>, UInt32) throws(Error) -> Result
+      _ body: (UnsafePointer<CInterop.SocketAddress>, CInterop.SocketLength) throws(Error) -> Result
     ) rethrows -> Result where Error: Swift.Error
     
     static func withUnsafePointer(
@@ -16,7 +16,7 @@ public protocol SocketAddress: Sendable {
     ) -> Self
     
     static func withUnsafePointer(
-        _ body: (UnsafeMutablePointer<CInterop.SocketAddress>, UInt32) throws -> ()
+        _ body: (UnsafeMutablePointer<CInterop.SocketAddress>, CInterop.SocketLength) throws -> ()
     ) rethrows -> Self
 }
 

--- a/Sources/Socket/System/SocketAddress/IPv4SocketAddress.swift
+++ b/Sources/Socket/System/SocketAddress/IPv4SocketAddress.swift
@@ -32,7 +32,7 @@ public struct IPv4SocketAddress: SocketAddress, Equatable, Hashable {
     }
     
     public func withUnsafePointer<Result, Error>(
-      _ body: (UnsafePointer<CInterop.SocketAddress>, UInt32) throws(Error) -> Result
+      _ body: (UnsafePointer<CInterop.SocketAddress>, CInterop.SocketLength) throws(Error) -> Result
     ) rethrows -> Result where Error: Swift.Error {
         
         var socketAddress = CInterop.IPv4SocketAddress()
@@ -51,7 +51,7 @@ public struct IPv4SocketAddress: SocketAddress, Equatable, Hashable {
     }
     
     public static func withUnsafePointer(
-        _ body: (UnsafeMutablePointer<CInterop.SocketAddress>, UInt32) throws -> ()
+        _ body: (UnsafeMutablePointer<CInterop.SocketAddress>, CInterop.SocketLength) throws -> ()
     ) rethrows -> Self {
         var socketAddress = CInterop.IPv4SocketAddress()
         try socketAddress.withUnsafeMutablePointer(body)

--- a/Sources/Socket/System/SocketAddress/IPv6SocketAddress.swift
+++ b/Sources/Socket/System/SocketAddress/IPv6SocketAddress.swift
@@ -32,7 +32,7 @@ public struct IPv6SocketAddress: SocketAddress, Equatable, Hashable {
     }
     
     public func withUnsafePointer<Result, Error>(
-      _ body: (UnsafePointer<CInterop.SocketAddress>, UInt32) throws(Error) -> Result
+        _ body: (UnsafePointer<CInterop.SocketAddress>, CInterop.SocketLength) throws(Error) -> Result
     ) rethrows -> Result where Error: Swift.Error {
         
         var socketAddress = CInterop.IPv6SocketAddress()
@@ -51,7 +51,7 @@ public struct IPv6SocketAddress: SocketAddress, Equatable, Hashable {
     }
     
     public static func withUnsafePointer(
-        _ body: (UnsafeMutablePointer<CInterop.SocketAddress>, UInt32) throws -> ()
+        _ body: (UnsafeMutablePointer<CInterop.SocketAddress>, CInterop.SocketLength) throws -> ()
     ) rethrows -> Self {
         var socketAddress = CInterop.IPv6SocketAddress()
         try socketAddress.withUnsafeMutablePointer(body)

--- a/Sources/Socket/System/SocketAddress/LinkLayerSocketAddress.swift
+++ b/Sources/Socket/System/SocketAddress/LinkLayerSocketAddress.swift
@@ -51,7 +51,7 @@ public struct LinkLayerSocketAddress: SocketAddress, Equatable, Hashable {
     }
     
     public func withUnsafePointer<Result, Error>(
-      _ body: (UnsafePointer<CInterop.SocketAddress>, UInt32) throws(Error) -> Result
+        _ body: (UnsafePointer<CInterop.SocketAddress>, CInterop.SocketLength) throws(Error) -> Result
     ) rethrows -> Result where Error: Swift.Error {
         
         var socketAddress = CSocketAddressType()
@@ -76,7 +76,7 @@ public struct LinkLayerSocketAddress: SocketAddress, Equatable, Hashable {
     }
     
     public static func withUnsafePointer(
-        _ body: (UnsafeMutablePointer<CInterop.SocketAddress>, UInt32) throws -> ()
+        _ body: (UnsafeMutablePointer<CInterop.SocketAddress>, CInterop.SocketLength) throws -> ()
     ) rethrows -> Self {
         var socketAddress = CSocketAddressType()
         try socketAddress.withUnsafeMutablePointer(body)

--- a/Sources/Socket/System/SocketAddress/UnixSocketAddress.swift
+++ b/Sources/Socket/System/SocketAddress/UnixSocketAddress.swift
@@ -27,7 +27,7 @@ public struct UnixSocketAddress: SocketAddress, Equatable, Hashable {
     }
     
     public func withUnsafePointer<Result, Error>(
-      _ body: (UnsafePointer<CInterop.SocketAddress>, UInt32) throws(Error) -> Result
+      _ body: (UnsafePointer<CInterop.SocketAddress>, CInterop.SocketLength) throws(Error) -> Result
     ) rethrows -> Result where Error: Swift.Error {
         return try path.withPlatformString { platformString in
             var socketAddress = CInterop.UnixSocketAddress()
@@ -43,7 +43,7 @@ public struct UnixSocketAddress: SocketAddress, Equatable, Hashable {
     }
     
     public static func withUnsafePointer(
-        _ body: (UnsafeMutablePointer<CInterop.SocketAddress>, UInt32) throws -> ()
+        _ body: (UnsafeMutablePointer<CInterop.SocketAddress>, CInterop.SocketLength) throws -> ()
     ) rethrows -> Self {
         var socketAddress = CInterop.UnixSocketAddress()
         try socketAddress.withUnsafeMutablePointer(body)

--- a/Sources/Socket/System/SocketOperations.swift
+++ b/Sources/Socket/System/SocketOperations.swift
@@ -179,7 +179,7 @@ extension SocketDescriptor {
     ) -> Result<(), Errno> {
         nothingOrErrno(retryOnInterrupt: retryOnInterrupt) {
             option.withUnsafeBytes { bufferPointer in
-                system_setsockopt(self.rawValue, Option.ID.optionLevel.rawValue, Option.id.rawValue, bufferPointer.baseAddress!, UInt32(bufferPointer.count))
+                system_setsockopt(self.rawValue, Option.ID.optionLevel.rawValue, Option.id.rawValue, bufferPointer.baseAddress!, CInterop.SocketLength(bufferPointer.count))
             }
         }
     }
@@ -208,7 +208,7 @@ extension SocketDescriptor {
     ) -> Result<Option, Errno> {
         do {
             let value = try Option.withUnsafeBytes { bufferPointer throws(Errno) -> () in
-                var length = UInt32(bufferPointer.count)
+                var length = CInterop.SocketLength(bufferPointer.count)
                 guard system_getsockopt(self.rawValue, Option.ID.optionLevel.rawValue, Option.id.rawValue, bufferPointer.baseAddress!, &length) != -1 else {
                     throw Errno.current
                 }
@@ -498,7 +498,7 @@ extension SocketDescriptor {
     internal func _accept(
         retryOnInterrupt: Bool
     ) -> Result<SocketDescriptor, Errno> {
-        var length: UInt32 = 0
+        var length: CInterop.SocketLength = 0
         return valueOrErrno(retryOnInterrupt: retryOnInterrupt) {
             system_accept(self.rawValue, nil, &length)
         }.map(SocketDescriptor.init(rawValue:))

--- a/Sources/Socket/System/Syscalls.swift
+++ b/Sources/Socket/System/Syscalls.swift
@@ -162,7 +162,7 @@ internal func system_inet_pton(
   return inet_pton(family, cString, address)
 }
 
-internal func system_inet_ntop(_ family: Int32, _ pointer : UnsafeRawPointer, _ string: UnsafeMutablePointer<CChar>, _ length: UInt32) -> UnsafePointer<CChar>? {
+internal func system_inet_ntop(_ family: Int32, _ pointer : UnsafeRawPointer, _ string: UnsafeMutablePointer<CChar>, _ length: socklen_t) -> UnsafePointer<CChar>? {
   #if ENABLE_MOCKING
   //if mockingEnabled { return _mock(family, pointer, string, length) }
   #endif
@@ -176,7 +176,7 @@ internal func system_socket(_ fd: Int32, _ fd2: Int32, _ fd3: Int32) -> Int32 {
   return socket(fd, fd2, fd3)
 }
 
-internal func system_setsockopt(_ fd: Int32, _ fd2: Int32, _ fd3: Int32, _ pointer: UnsafeRawPointer, _ dataLength: UInt32) -> Int32 {
+internal func system_setsockopt(_ fd: Int32, _ fd2: Int32, _ fd3: Int32, _ pointer: UnsafeRawPointer, _ dataLength: socklen_t) -> Int32 {
   #if ENABLE_MOCKING
   if mockingEnabled { return _mock(fd, fd2, fd3, pointer, dataLength) }
   #endif
@@ -188,7 +188,7 @@ internal func system_getsockopt(
   _ level: CInt,
   _ option: CInt,
   _ value: UnsafeMutableRawPointer?,
-  _ length: UnsafeMutablePointer<UInt32>
+  _ length: UnsafeMutablePointer<socklen_t>
 ) -> CInt {
   #if ENABLE_MOCKING
   if mockingEnabled { return _mock(socket, level, option, value, length) }
@@ -199,7 +199,7 @@ internal func system_getsockopt(
 internal func system_bind(
     _ socket: CInt,
     _ address: UnsafePointer<CInterop.SocketAddress>,
-    _ length: UInt32
+    _ length: socklen_t
 ) -> CInt {
   #if ENABLE_MOCKING
   if mockingEnabled { return _mock(socket, address, length) }
@@ -247,7 +247,7 @@ internal func system_getaddrinfo(
 
 internal func system_getnameinfo(
   _ sa: UnsafePointer<CInterop.SocketAddress>,
-  _ salen: UInt32,
+  _ salen: socklen_t,
   _ host: UnsafeMutablePointer<CChar>?,
   _ hostlen: socklen_t,
   _ serv: UnsafeMutablePointer<CChar>?,
@@ -325,7 +325,7 @@ internal func system_sendto(
   _ length: Int,
   _ flags: CInt,
   _ dest_addr: UnsafePointer<CInterop.SocketAddress>?,
-  _ dest_len: UInt32
+  _ dest_len: socklen_t
 ) -> Int {
   #if ENABLE_MOCKING
   if mockingEnabled {
@@ -341,7 +341,7 @@ internal func system_recvfrom(
   _ length: Int,
   _ flags: CInt,
   _ address: UnsafeMutablePointer<CInterop.SocketAddress>?,
-  _ addres_len: UnsafeMutablePointer<UInt32>?
+  _ addres_len: UnsafeMutablePointer<socklen_t>?
 ) -> Int {
   #if ENABLE_MOCKING
   if mockingEnabled {
@@ -509,14 +509,14 @@ internal func system_link_ntoa(_ address: UnsafePointer<sockaddr_dl>) -> UnsafeM
 }
 #endif
 
-internal func system_getsockname(_ fd: CInt, _ address: UnsafeMutablePointer<CInterop.SocketAddress>, _ length: UnsafeMutablePointer<UInt32>) -> CInt {
+internal func system_getsockname(_ fd: CInt, _ address: UnsafeMutablePointer<CInterop.SocketAddress>, _ length: UnsafeMutablePointer<socklen_t>) -> CInt {
 #if ENABLE_MOCKING
     if mockingEnabled { return _mock(fd, address) }
 #endif
     return getsockname(fd, address, length)
 }
 
-internal func system_getpeername(_ fd: CInt, _ address: UnsafeMutablePointer<CInterop.SocketAddress>, _ length: UnsafeMutablePointer<UInt32>) -> CInt {
+internal func system_getpeername(_ fd: CInt, _ address: UnsafeMutablePointer<CInterop.SocketAddress>, _ length: UnsafeMutablePointer<socklen_t>) -> CInt {
 #if ENABLE_MOCKING
     if mockingEnabled { return _mock(fd, address) }
 #endif

